File: backend/apps/rides/serializers/ride.py
from rest_framework import serializers
from apps.rides.models import Ride


class CreateRideSerializer(serializers.Serializer):
    pickup_lat = serializers.FloatField()
    pickup_lng = serializers.FloatField()
    pickup_address = serializers.CharField(max_length=255)
    dropoff_lat = serializers.FloatField()
    dropoff_lng = serializers.FloatField()
    dropoff_address = serializers.CharField(max_length=255)


class RideSerializer(serializers.ModelSerializer):
    class Meta:
        model = Ride
        fields = [
            "id", "status", "pickup_address", "dropoff_address",
            "estimated_price", "final_price", "estimated_distance",
            "estimated_duration", "created_at", "completed_at",
        ]
        read_only_fields = fields

File: admin-panel/src/types/ride.ts
export type RideStatus = 'searching' | 'accepted' | 'arrived' | 'in_progress' | 'completed' | 'cancelled';

export interface Ride {
  id: string;
  status: RideStatus;
  pickup_address: string;
  dropoff_address: string;
  estimated_price: string;
  final_price: string | null;
  estimated_distance: number;
  estimated_duration: number;
  created_at: string;
  completed_at: string | null;
}
